Header.c

```diff
@@ -46,6 +46,16 @@ void Header_populateFromSettings(Header* this) {
    Header_forEachColumn(this, col) {
       const MeterColumnSettings* colSettings = &this->settings->columns[col];
       for (int i = 0; i < colSettings->len; i++) {
+         char* end, dynamic[32];
+         unsigned int param = 0;
+         if(String_startsWith(colSettings->names[i], "Dynamic")) {
+            char* paren = strchr(colSettings->names[i], '(');
+            int ok = sscanf(paren, "(%30s)", dynamic); // DynamicMeter
+            if (ok && (end = strrchr(dynamic, ')'))) *end = '\0';
+            param = ok ? DynamicMeter_search(this->pl, dynamic) : 0;
+            if(!param)
+               continue;
+         }
          Header_addMeterByName(this, colSettings->names[i], col);
```

Comment:
Good catch and good fix @smalinux - I think though that this way of tackling it ends up duplicating parsing code that's inside Header_addMeterByName() at line 100 already.  It'd be simpler to change that line to return a MeterModeId that indicates the error (e.g. LAST) and then make the calling code just skip onto the next loop entry.

I tend to agree this validation needs to happen before the call to Meter_new though (which could separately acquire some validation as @BenBE suggests, but maybe not for this issue - we'll never even reach Meter_new in this case).  I'll send through a revised patch.

I see there's another problematic situation, kinda the inverse to this one @smalinux - where we could have duplicate definitions of a DynamicMeter (e.g. in /etc/pcp/htop and ~/.config/htop/htoprc) - we'll need a diagnostic to warn about that I think (and pick one based on some priority ordering).  At the moment it causes an error because we attempt to define the same derived metric twice.  Anyway, fix incoming.